cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu.go

```diff
@@ -1025,31 +1060,84 @@ func mergeKubeconfigAndSetContext(mgr kubeconfig.Manager, kcPath string) error {
 	return nil
 }
 
-// resolveCNI determines which CNI package to use. It expects to be passed a
-// fully qualified package name except for special known CNI values such as
-// antrea or calico.
-func resolveCNI(mgr packages.PackageManager, cniName string) (*CNIPackage, error) {
-	if cniName == cniNoneName {
-		return nil, fmt.Errorf("CNI was set to %s", cniName)
+// resolvePkg picks the first package in the package repo
+// that matches the name and version of the provided profile
+// If the user did not specify a profile version, defaults to the first one found which should be the latest
+func resolvePkg(mgr packages.PackageManager, namespace, pkgName, pkgVersion string) (*Package, error) {
+	if pkgName == cniNoneName {
+		log.Style(outputIndent, color.FgYellow).Warnf("No CNI installed: CNI was set to %s.\n", cniNoneName)
+		return nil, nil
 	}
-	pkgs, err := mgr.ListPackagesInNamespace(tkgSysNamespace)
+
+	pkgs, err := mgr.ListPackagesInNamespace(namespace)
 	if err != nil {
 		return nil, err
 	}
 
-	cniPkg := &CNIPackage{}
+	versions := []string{}
+
+	profilePkg := &Package{
+		installName: pkgName,
+	}
+
 	for _, pkg := range pkgs {
-		if strings.HasPrefix(pkg.Spec.RefName, cniName) {
-			cniPkg.fqPkgName = pkg.Spec.RefName
-			cniPkg.pkgVersion = pkg.Spec.Version
+		// Select the package by name directly or by a well known prefix (like calico, antrea, fluent-bit)
+		if pkg.Spec.RefName == pkgName || strings.HasPrefix(pkg.Spec.RefName, pkgName) {
+			if pkgVersion == "" || pkg.Spec.Version == pkgVersion {
+				profilePkg.fqPkgName = pkg.Spec.RefName
+				profilePkg.pkgVersion = pkg.Spec.Version
+
+				// Build list of semantic versions for matching package
+				versions = append(versions, pkg.Spec.Version)
+			}
 		}
 	}
+
+	if profilePkg.fqPkgName == "" {
+		return nil, fmt.Errorf("no package was resolved for named %s with version %s", pkgName, pkgVersion)
+	}
+
+	// sort and select latest semver (last in slice) if user did not specify the version to use
+	if pkgVersion == "" {
+		semver.Sort(versions)
+		profilePkg.pkgVersion = versions[len(versions)-1]
+	}
+
+	return profilePkg, nil
+}
+
+// installProfile installs the profile package to be satisfied via kapp-controller and any provided config file
+func installProfile(pkgClient packages.PackageManager, t *UnmanagedCluster, profileConfigPath string) error {
+	rootSvcAcct, err := pkgClient.GetRootServiceAccount(tkgSysNamespace, tkgSvcAcctName)
 	if err != nil {
-		return nil, err
+		log.Errorf("failed to get root service account: %s\n", err.Error())
+		return err
 	}
```

Comment:
Below here, we need to check if `rootSvcAcct` is nil, if it is, return an error.

Otherwise, should that condition ever happen, we'll panic below when you call `rootSvcAcct.Name`.